docs/nomad/README.md

```diff
@@ -0,0 +1,262 @@
+        # Get the latest version on https://hub.docker.com/r/hetznercloud/hcloud-csi-driver/tags
+        image      = "hetznercloud/hcloud-csi-driver:v2.3.2"
+        command    = "bin/hcloud-csi-driver-node"
+        privileged = true
+      }
+
+      env {
+        CSI_ENDPOINT   = "unix://csi/csi.sock"
+        ENABLE_METRICS = true
+      }
+
+      template {
+        data        = <<EOH
+HCLOUD_TOKEN="{{ with nomadVar "secrets/hcloud" }}{{ .hcloud_token }}{{ end }}"
+EOH
+        destination = "${NOMAD_SECRETS_DIR}/hcloud-token.env"
+        env         = true
+      }
+
+      csi_plugin {
+        id        = "csi.hetzner.cloud"
+        type      = "node"
+        mount_dir = "/csi"
+      }
+
+      resources {
+        cpu    = 100
+        memory = 64
+      }
+    }
+  }
+}
+```
+
+5. Deploy the Jobs:
+
+The following commands deploy the job resources created previously on your Nomad cluster:
+
+```sh
+nomad job run hcloud-csi-controller.hcl
+nomad job run hcloud-csi-node.hcl
+```
+
+6. Verify the status:
+
+To ensure the plugin is running and healthy, check the web UI on path `/ui/csi/plugins/csi.hetzner.cloud` or by using the CLI:
+
+```sh
+nomad plugin status
+```
+
+### Volumes Setup
+
+1. Define a Volume:
+
+Create a file `db-vol.hcl` for the volume resource:
+
+```
+# file: db-vol.hcl
+
+type      = "csi"
+id        = "db-vol"
+name      = "db-vol"
+namespace = "default"
+plugin_id = "csi.hetzner.cloud"
+
+capability {
+  access_mode     = "single-node-writer"
+  attachment_mode = "file-system"
+}
+
+mount_options {
+  fs_type     = "ext4"
+  mount_flags = ["discard", "defaults"]
+}
+```
+
+2. Create a Volume:
+
+```sh
+nomad volume create db-vol.hcl
+```
+
+> [!HINT]
+>  The hcloud cli provides a convient way to verify if the volume was created: `hcloud volume list`.
+
+### Make use of the Volume
+
+1. Create a Job definition:
+
+The following example describes how to mount the volume in a Docker Nomad job definition:
+
+```hcl
+# file: mariadb.nomad
+
+job "mariadb" {
+  datacenters = ["dc1"]
+  namespace   = "default"
+  type        = "service"
+
+  group "mariadb" {
+    network {
+      port "mariadb" {
+        to           = 3306
+      }
+    }
+
+    volume "db-volume" {
+      type            = "csi"
+      read_only       = false
+      source          = "db-vol"
+      attachment_mode = "file-system"
+      access_mode     = "single-node-writer"
+      per_alloc       = false
+    }
+
+    task "mariadb" {
+      driver = "docker"
+      config {
+        image = "mariadb:10.11"
+        ports = [
+          "mariadb",
+        ]
+      }
+
+      volume_mount {
+        volume      = "db-volume"
+        destination = "/var/lib/mysql"
+      }
+
+      env {
+        MARIADB_ROOT_PASSWORD = "<...>"
+        MARIADB_DATABASE      = "<...>"
+        MARIADB_USER          = "<...>"
+        MARIADB_PASSWORD      = "<...>"
+      }
+
+      service {
+        name = "db"
+        port = "mariadb"
+      }
```

Comment:
good point, since Nomad has a "lightweight" native service discovery (https://www.hashicorp.com/blog/nomad-service-discovery), let's use this instead and add a note about consul.

Reply:
see 48a51c65aac71c5f231fd27c23873f50f1543f20